methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_map(proc = nil)
+    temp = []
+    i = 0
+    if proc.nil?
+      length.times do
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    else
+      length.times do
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    length.times do
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    length.times do
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
- [ ] `my_all` is not working as the original `all?` method, also is not passing all test cases. 

### `all?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are not false.
```
[1, 'string', true].all?
# true
```
- Returns `false` if it finds at least one element that is not true
```
[1, 'string', nil].all?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition
- it can use a block but it is not necessary to make the method work

[Enumerable#all?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-all-3F)